ush/soca/marine_recenter.py

```diff
@@ -155,6 +161,37 @@ def initialize(self):
                 ens_member_list.append([fname_in, fname_out])
 
         FileHandler({'copy': ens_member_list}).sync()
+        # stage ensemble ice restarts
+        # make a copy of the CICE6 restart
+        logger.info("---------------- Stage ensemble CICE restarts")
+        # set the restart date, dependent on the cycling type
+        if self.task_config.DOIAU:
+            # forecast initialized at the begining of the DA window
+            fcst_begin = self.task_config.MARINE_WINDOW_BEGIN_ISO
+            rst_date = self.task_config.MARINE_WINDOW_BEGIN.strftime('%Y%m%d.%H%M%S')
+        else:
+            # forecast initialized at the middle of the DA window
+            fcst_begin = self.task_config.MARINE_WINDOW_MIDDLE_ISO
+            rst_date = self.task_config.MARINE_WINDOW_MIDDLE.strftime('%Y%m%d.%H%M%S')
+        ens_cice_list = []
+        for mem in range(1, nmem_ens+1):
+            mem_dir = os.path.join(self.task_config.ROTDIR,
+                                   f'enkf{RUN}.{gPDYstr}',
+                                   f'{gcyc}',
+                                   f'mem{str(mem).zfill(3)}',
+                                   'model',
+                                   'ice',
+                                   'restart')
+            mem_dir_real = os.path.realpath(mem_dir)
+            f00 = f'{rst_date}.cice_model.res.nc'
+            fname_in = os.path.abspath(os.path.join(mem_dir_real, f00))
+            fname_out = os.path.realpath(os.path.join(self.task_config.ens_dir,
+                                             "cice_model.res."+str(mem)+".nc"))
+            ens_cice_list.append([fname_in, fname_out])
+            fname_out = os.path.realpath(os.path.join(self.task_config.ens_dir,
+                                             "cice_model.res.output."+str(mem)+".nc"))
```

Comment:
Note: this currently copies restarts twice (to keep input and output separate), we can change this before the merge to only copy the file once, and change it in place. (I coded input and output separately so I can easily see what the recentering does).